src/app/main/listing/listing.component.ts

```diff
@@ -39,8 +57,13 @@ export class ListingComponent implements OnInit, OnDestroy {
           return this.listingService.getOneListing(listingId);
         }),
         tap(listing => {
-          this.listingForm.get('listing').patchValue(listing);
-          this.listingForm.get('listingType').patchValue(listing ? listing.listingType : null);
+          if (listing) {
+            this.formGroup.setValue(this.transformToFormGroup(listing));
```

Comment:
`this.transformToFormGroup` is that needed?

Reply:
Yep as this form is another case of having polymorphic & therefore different data structures for the from.